src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let res = this[0];
+
+    if (!this.length) {
+      return '';
+    }
+
+    if (res === null) {
+      res = '';
+    }
```

Comment:
This is redudant if fix `if` condition
```suggestion
    let res = '';
```